File: backend/src/main/java/com/artifax_solutions/website/model/BlogHubSection.java
package com.artifax_solutions.website.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class BlogHubSection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String subTitle;
    private String title;
    private String description;
    private String buttonText;
    private String buttonUrl;
    private boolean showNavigationArrows;
    private boolean showNavigationIndicators;
    @OneToMany(mappedBy = "section", cascade = CascadeType.ALL)
    private List<BlogHubItem> itemList;
}

File: frontend/src/interface/BlogHubSection.ts
export interface BlogHubSection {
  id: number | null | undefined;
  subTitle: string | null | undefined;
  title: string | null | undefined;
  description: string | null | undefined;
  buttonText: string | null | undefined;
  buttonUrl: string | null | undefined;
  showNavigationArrows: boolean;
  showNavigationIndicators: boolean;
  itemList: BlogHubItem[];
}

export interface BlogHubItem {
  id: number | undefined | null;
  imageUrl: string | undefined | null;
  altText: string | undefined | null;
  title: string | undefined | null;
  content: string | undefined | null;
  linkUrl: string | undefined | null;
}

